text/trace/0205-messaging-semantic-conventions-context-propagation.md

```diff
@@ -0,0 +1,173 @@
+created by the producer and must be propagated to the consumers. It must not be
+altered by intermediaries.  This context helps to model dependencies between
+producers and consumers, regardless of the underlying messaging transport
+mechanism and its instrumentation.
+
+Instrumentors are required to instrument producer and consumer applications
+so that context is attached to messages and extracted from messages in a
+coordinated way. Future versions of these conventions might recommend [context propagation according to certain industry standards](#standards-for-context-propagation).
+
+### Requirements
+
+A producer SHOULD attach a creation context to each message. The creation context
+SHOULD be attached in a way so that it is not possible to be changed by intermediaries.
+
+## Future possibilities
+
+### Transport context propagation
+
+As illustrated in the [example](#example) above, a creation context can be
+attached to a message, while different contexts are propagated via HTTP requests
+publishing and fetching the message. When coming up with conventions and
+guidance for intermediary instrumentation, it will be beneficial to clearly
+outline those two layers of context propagation and build conventions for
+intermediary instrumentation on top of this outline:
+
+1. The _creation context layer_ allows correlating the producer with the
+   consumers of a message, regardless of intermediary instrumentation. The
+   creation context is created by the producer and must be propagated to the
+   consumers. It must not be altered by intermediaries.
+
+   This layer helps to model dependencies between producers and consumers,
+   regardless of the underlying messaging transport mechanism and its
+   instrumentation.
+2. An additional _transport context layer_ allows correlating the producer and
+   the consumer with an intermediary. It also allows correlating multiple
+   intermediaries among each other. The transport context can be changed by
+   intermediaries, according to intermediary instrumentations. Intermediaries that
+   are not instrumented might simply drop the transport context.
+
+   This layer helps to gain insights into details of the message transport.
+
+This would keep the existing correlation between producers and consumers intact
+while allowing intermediaries to use the transport context to correlate
+intermediary instrumentation with existing producer and consumer
+instrumentations.
+
+### Standards for context propagation
+
+Currently, instrumentors have to decide how to attach and extract context from
+messages to fulfil the [requirements for context propagation](#context-propagation).
+While preserving the freedom for instrumentors to choose how to propagate
+context, in the future these conventions should list recommended ways of how to
+propagate context using popular messaging protocols.
+
+Currently, several attempts exist to standardize context propagation for different
```

Comment:
```suggestion
There are several working-in-progress efforts to standardize context propagation for different
```
Maybe?

Reply:
This way, we could also drop the  `Those standards are in draft states and/or are not widely adopted yet` part below, since it's implicit that if they are "work-in-progress" that they are drafts and not adopted.